src/main/java/org/kiwiproject/registry/util/Ports.java

```diff
@@ -57,7 +57,6 @@ public static Port findPort(List<Port> ports, Port.Security security, Port.PortT
      */
     public static String determineScheme(List<Port> ports, Port.PortType type) {
         var firstPort = findFirstPortPreferSecure(ports, type);
```

Comment:
Question: Does this ever return `null`??? (I am guessing 'no' but we'd get a lovely NPE if it can...)

Reply:
No it can not return null. If a port is not found, a dummy object is made with port zero and the type and security passed in.